docker/build/Dockerfile

```diff
@@ -17,7 +16,7 @@ COPY --from=goreleaser/goreleaser:v1.12.3 /usr/bin/goreleaser /usr/local/bin/
 
 COPY --from=alpine/helm:3.12.2 /usr/bin/helm /usr/local/bin/
 
-COPY --from=golangci/golangci-lint:v1.50 /usr/bin/golangci-lint /usr/local/bin/
+COPY --from=golangci/golangci-lint:v1.51.2 /usr/bin/golangci-lint /usr/local/bin/
```

Comment:
Why do we need to upgrade `golangci-lint`? If we need to upgrade, we should switch to the latest version [1.54.2](https://github.com/golangci/golangci-lint/releases/tag/v1.54.2)

Reply:
I tried with 1.54.2 but it was reporting false positives due to the `unparam` linter hence had to go back.
```
pkg/controllers/repositoryserver/handler.go:344:98: (*RepoServerHandler).setupKopiaRepositoryServer - result 0 (sigs.k8s.io/controller-runtime/pkg/reconcile.Result) is always nil (unparam)
func (h *RepoServerHandler) setupKopiaRepositoryServer(ctx context.Context, logger logr.Logger) (ctrl.Result, error) {

```

Reply:
Sounds good! Could you please create a github issue and add a link here. We can resolve the conversation after that

Reply:
https://github.com/kanisterio/kanister/issues/2408